File: Labtwo/PHP/pr3.php
<?php
function funcnum($numb) {
    // Проверка что все символы - цифры
    if (!ctype_digit($numb)) {
        return;
    }
    
    // Проверка длины
    if (strlen($numb) < 2) {
        return;
    }
    
    $num = intval($numb);
    $sum = 0;
    $proizv = 1;
    
    echo "$num: ";
    
    // Сумма цифр
    for ($i = 0; $i < strlen($numb); $i++) {
        $d = intval($numb[$i]);
        $sum += $d;
        echo $numb[$i];
        if ($i < strlen($numb) - 1) {
            echo "+";
        }
    }
    echo "=$sum, ";
    
    // Произведение цифр
    for ($i = 0; $i < strlen($numb); $i++) {
        $d = intval($numb[$i]);
        $proizv *= $d;
        echo $numb[$i];
        if ($i < strlen($numb) - 1) {
            echo "*";
        }
    }
    echo "=$proizv, ";
    
    // Модуль разности
    $diff = abs($sum - $proizv);
    echo "|$sum-$proizv|=$diff\n";
}

echo "Write your number: ";
$input = trim(fgets(STDIN));

// Обработка каждого числа
foreach (explode(' ', $input) as $numb) {
    funcnum($numb);
}
?>

File: Labtwo/GO/pr3.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func funcnum(numb string) {
	// Проверка что все символы - цифры
	for _, c := range numb {
		if !(c >= '0' && c <= '9') {
			return
		}
	}

	// Проверка длины
	if len(numb) < 2 {
		return
	}

	// Парсинг числа
	num, _ := strconv.Atoi(numb)
	sum := 0
	proizv := 1

	// Вывод числа
	fmt.Printf("%d: ", num)

	// Вычисление суммы
	for i, c := range numb {
		d := int(c - '0')
		sum += d
		fmt.Printf("%c", c)
		if i < len(numb)-1 {
			fmt.Print("+")
		}
	}
	fmt.Printf("=%d, ", sum)

	// Вычисление произведения
	for i, c := range numb {
		d := int(c - '0')
		proizv *= d
		fmt.Printf("%c", c)
		if i < len(numb)-1 {
			fmt.Print("*")
		}
	}
	fmt.Printf("=%d, ", proizv)

	// Модуль разности
	diff := sum - proizv
	if diff < 0 {
		diff = -diff
	}
	fmt.Printf("|%d-%d|=%d\n", sum, proizv, diff)
}

func main() {
	fmt.Println("Write your number")
	var input string
	fmt.Scanln(&input)

	// Разделение ввода по пробелам
	for _, numb := range strings.Fields(input) {
		funcnum(numb)
	}
}